tests/client/test_client.py

```diff
@@ -307,9 +307,11 @@ def test_run_task_sets_up_control(
 ):
     mock_rest.create_task.return_value = TaskResponse(task_id="foo")
     mock_rest.update_worker_task.return_value = TaskResponse(task_id="foo")
+    ctx = Mock()
+    ctx.correlation_id = "foo"
+    mock_events.subscribe_to_all_events = lambda on_event: on_event(COMPLETE_EVENT, ctx)
```

Comment:
I had to make this change to ensure that the task completes and doesn't hang indefinitely. I previously tried setting a short timeout, which allowed the test to finish, but it introduced an unintended dependency on `test_valid_stomp_config_for_listener` test. This is not ideal, as tests should be independent of each other. It seems the issue is related to the task handling.

```
        == "Subscribing to all bluesky events from localhost:61613\nPress enter to "
        + 'exit{\n  "state": "IDLE",\n  "task_status": null,\n  "errors": [],\n  '
        + '"warnings": []\n}\n'
```